embulk-core/src/main/java/org/embulk/spi/Row.java

```diff
@@ -0,0 +1,318 @@
+package org.embulk.spi;
+
+import org.embulk.spi.time.Timestamp;
+import org.embulk.spi.type.Type;
+import org.embulk.spi.type.Types;
+import org.msgpack.value.Value;
+
+/**
+ * This class is a temporal container for storing a row only.
+ */
+class Row
+{
+    static Row newRow(Schema schema)
+    {
+        Val[] values = new Val[schema.getColumnCount()];
+        for (Column column : schema.getColumns()) {
+            values[column.getIndex()] = newValue(column);
+        }
+        return new Row(values);
+    }
+
+    private static Val newValue(Column column)
+    {
+        Type type = column.getType();
+        if (type.equals(Types.BOOLEAN)) {
+            return new BooleanVal(column);
+        }
+        else if (type.equals(Types.DOUBLE)) {
+            return new DoubleVal(column);
+        }
+        else if (type.equals(Types.LONG)) {
+            return new LongVal(column);
+        }
+        else if (type.equals(Types.STRING)) {
+            return new StringVal(column);
+        }
+        else if (type.equals(Types.JSON)) {
+            return new JsonVal(column);
+        }
+        else if (type.equals(Types.TIMESTAMP)) {
+            return new TimestampVal(column);
+        }
+        else {
+            throw new IllegalStateException("Unsupported type " + type.getName());
+        }
+    }
+
+    private final Val[] values;
+
+    private Row(Val[] values)
+    {
+        this.values = values;
+    }
+
+    void setNull(int columnIndex)
+    {
+        values[columnIndex].setNull();
+    }
+
+    void setBoolean(int columnIndex, boolean value)
+    {
+        values[columnIndex].setBoolean(value);
+    }
+
+    void setLong(int columnIndex, long value)
+    {
+        values[columnIndex].setLong(value);
+    }
+
+    void setDouble(int columnIndex, double value)
+    {
+        values[columnIndex].setDouble(value);
+    }
+
+    void setString(int columnIndex, String value)
+    {
+        values[columnIndex].setString(value);
+    }
+
+    void setJson(int columnIndex, Value value)
+    {
+        values[columnIndex].setJson(value);
+    }
+
+    void setTimestamp(int columnIndex, Timestamp value)
+    {
+        values[columnIndex].setTimestamp(value);
+    }
+
+    void write(PageBuilder pageBuilder)
+    {
+        for (Val v : values) {
+            v.write(pageBuilder);
+        }
+    }
+
+    interface Val
```

Comment:
Let's use `Value` instead of `Val`. I don't think we achieve any benefit by this abbreviation.